### PR TITLE
🐛 bugfix: model_builder `search` config might be None

### PR DESCRIPTION
**Pull request by trajepl**
## Describe your changes

model_build `search` config might be None

## Checklist before requesting a review
- [ ] Add unit tests for this change.
- [ ] Make sure all tests can pass.
- [ ] Update documents if necessary.
- [ ] Lint and apply fixes to your code by running `lintrunner -a`
- [ ] Is this a user-facing change? If yes, give a description of this change to be included in the release notes.
- [ ] Is this PR including examples changes? If yes, please remember to update [example documentation](https://github.com/microsoft/Olive/blob/main/docs/source/examples.md) in a follow-up PR.

### CODE REVIEW

File: olive/passes/onnx/model_builder.py [hunk 1]
```diff
@@ -216,7 +216,9 @@ def _run_for_config(
         with open(genai_config_filepath) as istrm:
             genai_config = json.load(istrm)
 
-        genai_config["search"] = {**genai_config.get("search", {}), **config.get("search", {})}
+        new_search_options = config.get("search") or {}
+        new_search_options.update(genai_config.get("search") or {})
```
Comment (jambayk) [inline olive/passes/onnx/model_builder.py:220]: I think the order the opposite. we want `genai_config.update(config.get("search) or {})`

Comment (jambayk) [reply olive/passes/onnx/model_builder.py:220]: btw, python doesn't complain about common keys in the `{**a, **b}` solution 
image

`genai_config["search"] = {**genai_config.get("search", {}), **(config.get("search") or {})}` also works.

File: olive/passes/onnx/model_builder.py [hunk 2]
```diff
@@ -216,7 +216,9 @@ def _run_for_config(
         with open(genai_config_filepath) as istrm:
             genai_config = json.load(istrm)
 
-        genai_config["search"] = {**genai_config.get("search", {}), **config.get("search", {})}
+        new_search_options = config.get("search") or {}
```
Comment (guotuofeng) [inline olive/passes/onnx/model_builder.py:219]: don't change the original config

File: olive/passes/onnx/model_builder.py (hunk 1 above)
Comment (trajepl) [reply olive/passes/onnx/model_builder.py]: You are right. Updated like:
`genai_config["search"] = {**(genai_config.get("search") or {}), **(config.get("search") or {})}`

to ensure it is safe to merge genai_config, config's search config.

**Review by jambayk — APPROVED** [review]

**Pull request merged**